query/src/main/java/org/infinispan/query/dsl/embedded/impl/LuceneQueryMaker.java

```diff
@@ -217,10 +213,13 @@ public Query visit(LikeExpr likeExpr) {
       PropertyValueExpr propertyValueExpr = (PropertyValueExpr) likeExpr.getChild();
       String[] propertyPath = propertyValueExpr.getPropertyPath();
       ensureNotAnalyzed(propertyPath);
-      String patternValue = MULTIPLE_CHARACTERS_WILDCARD_PATTERN.matcher(likeExpr.getPattern()).replaceAll(LUCENE_MULTIPLE_CHARACTERS_WILDCARD);
-      patternValue = SINGLE_CHARACTER_WILDCARD_PATTERN.matcher(patternValue).replaceAll(LUCENE_SINGLE_CHARACTER_WILDCARD);
+      String lucenePattern = likeExpr.getPattern()
+            .replace(LUCENE_MULTIPLE_CHARACTERS_WILDCARD, "\\" + LUCENE_MULTIPLE_CHARACTERS_WILDCARD)
+            .replace(LUCENE_SINGLE_CHARACTER_WILDCARD, "\\" + LUCENE_SINGLE_CHARACTER_WILDCARD)
+            .replace("%", LUCENE_MULTIPLE_CHARACTERS_WILDCARD)
+            .replace("_", LUCENE_SINGLE_CHARACTER_WILDCARD);
```

Comment:
I think this would perform better with a single loop through the characters and a switch statement.